test/compress/let.js

```diff
@@ -381,6 +381,86 @@ reduce_vars: {
     node_version: ">=4"
 }
 
+reduce_vars_2: {
+    options = {
+        reduce_funcs: true,
+        reduce_vars: true,
+        unused: true,
+    }
+    input: {
+        "use strict";
+        (function() {
+            function f() {
+                console.log(typeof a);
+            }
+            for (let a in [ 42 ])
+                f();
+        })();
+    }
+    expect: {
+        "use strict";
+        (function() {
+            function f() {
+                console.log(typeof a);
+            }
+            for (let a in [ 42 ])
+                f();
+        })();
+    }
+    expect_stdout: "undefined"
+    node_version: ">=4"
+}
+
+reduce_vars_3: {
+    options = {
+        reduce_funcs: true,
+        reduce_vars: true,
+        unused: true,
+    }
+    input: {
+        "use strict";
+        (function(scope) {
+            let i = 1;
+            function f() {
+                i = 0;
+            }
+            for (let i = 0, x = 0; i < scope.length; i++, x++) {
+                if (x != i) {
+                    console.log("FAIL");
+                    break;
+                }
+                f();
+                console.log(scope[i]);
+            }
+            console.log(i);
+        })([ 4, 2 ]);
```

Comment:
@kzc this is your test case from https://github.com/mishoo/UglifyJS/pull/4776#issuecomment-799787702